code/modules/xenomorph/xeno_structures.dm

```diff
@@ -1534,7 +1534,7 @@ TUNNEL
 		return TRUE
 
 	var/mob/living/carbon/xenomorph/X = user
-	if(isxenoravager(X)) //Ask if this should be made into a trait for xenos with special ressources
+	if(!(X.xeno_caste.can_flags & CASTE_CAN_BE_GIVEN_PLASMA)) //Ask if this should be made into a trait for xenos with special ressources
```

Comment:
You're probably good to remove the comment, and you should change the flavor text if you can due to it being more themed around Ravagers and their rage resource.